### ISSUE TITLE
TestResultsFormatter: AssertionError: Dicts diff with key (display_name)

### ISSUE THREAD

**Issue by jonc101** [HealthRex/CDSS#120]
Just ran them to be sure Rocky could add to, but looks like a bunch of stuff does not pass:

======================================================================
ERROR: medinfo.dataconversion.test.TestEventDigraph (unittest.loader.ModuleImportFailure)
----------------------------------------------------------------------
ImportError: Failed to import test module: medinfo.dataconversion.test.TestEventDigraph
Traceback (most recent call last):
  File "C:\Dev\Python27\lib\unittest\loader.py", line 254, in _find_tests
    module = self._get_module_from_name(name)
  File "C:\Dev\Python27\lib\unittest\loader.py", line 232, in _get_module_from_name
    __import__(name)
  File "C:\HealthRex\CDSS\medinfo\dataconversion\test\TestEventDigraph.py", line 6, in <module>
    import networkx as nx
ImportError: No module named networkx

======================================================================
ERROR: test_newExtractorOutput (scripts.LabTestAnalysis.machine_learning.dataExtraction.testNewExtractorOutput.TestOutput)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\HealthRex\CDSS\scripts\LabTestAnalysis\machine_learning\dataExtraction\testNewExtractorOutput.py", line 18, in test_newExtractorOutput
    firstFile = gzip.open("labFeatureMatrix.LABFER.2013-01-01.2013-01-02.tab.gz",'r').read(12000)
  File "C:\Dev\Python27\lib\gzip.py", line 34, in open
    return GzipFile(filename, mode, compresslevel)
  File "C:\Dev\Python27\lib\gzip.py", line 94, in __init__
    fileobj = self.myfileobj = __builtin__.open(filename, mode or 'rb')
IOError: [Errno 2] No such file or directory: 'labFeatureMatrix.LABFER.2013-01-01.2013-01-02.tab.gz'

======================================================================
FAIL: test_addTimeCycleFeatures (medinfo.dataconversion.test.TestFeatureMatrixFactory.TestFeatureMatrixFactory)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\HealthRex\CDSS\medinfo\dataconversion\test\TestFeatureMatrixFactory.py", line 447, in test_addTimeCycleFeatures
    self.assertEqualList(resultMatrix, expectedMatrix)
  File "C:\HealthRex\CDSS\medinfo\common\test\Util.py", line 142, in assertEqualList
    self.assertEqual(verifyItem, sampleItem)
AssertionError: Lists differ: ['# feature-matrix_2018-04-20_... != ['pat_id', 'order_proc_id', 'p...

First differing element 0:
'# feature-matrix_2018-04-20_12.40.56.tab'
'pat_id'

Second list contains 10 additional elements.
First extra element 1:
'order_proc_id'

- ['# feature-matrix_2018-04-20_12.40.56.tab']
+ ['pat_id',
+  'order_proc_id',
+  'proc_code',
+  'order_time',
+  'normal_results',
+  'order_time.month',
+  'order_time.month.sin',
+  'order_time.month.cos',
+  'order_time.hour',
+  'order_time.hour.sin',
+  'order_time.hour.cos']

======================================================================
FAIL: test_buildFeatureMatrix_multiClinicalItem (medinfo.dataconversion.test.TestFeatureMatrixFactory.TestFeatureMatrixFactory)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\HealthRex\CDSS\medinfo\dataconversion\test\TestFeatureMatrixFactory.py", line 227, in test_buildFeatureMatrix_multiClinicalItem
    self.assertEqualList(resultMatrix, expectedMatrix)
  File "C:\HealthRex\CDSS\medinfo\common\test\Util.py", line 142, in assertEqualList
    self.assertEqual(verifyItem, sampleItem)
AssertionError: Lists differ: ['# feature-matrix_2018-04-20_... != ['pat_id', 'order_proc_id', 'p...

First differing element 0:
'# feature-matrix_2018-04-20_12.40.57.tab'
'pat_id'

Second list contains 56 additional elements.
First extra element 1:
'order_proc_id'

Diff is 1516 characters long. Set self.maxDiff to None to see it.

======================================================================
FAIL: test_buildFeatureMatrix_multiFlowsheet (medinfo.dataconversion.test.TestFeatureMatrixFactory.TestFeatureMatrixFactory)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\HealthRex\CDSS\medinfo\dataconversion\test\TestFeatureMatrixFactory.py", line 405, in test_buildFeatureMatrix_multiFlowsheet
    self.assertEqualList(resultMatrix, expectedMatrix)
  File "C:\HealthRex\CDSS\medinfo\common\test\Util.py", line 142, in assertEqualList
    self.assertEqual(verifyItem, sampleItem)
AssertionError: Lists differ: ['# feature-matrix_2018-04-20_... != ['pat_id', 'order_proc_id', 'p...

First differing element 0:
'# feature-matrix_2018-04-20_12.40.57.tab'
'pat_id'

Second list contains 49 additional elements.
First extra element 1:
'order_proc_id'

Diff is 1559 characters long. Set self.maxDiff to None to see it.

======================================================================
FAIL: test_buildFeatureMatrix_prePostFeatures (medinfo.dataconversion.test.TestFeatureMatrixFactory.TestFeatureMatrixFactory)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\HealthRex\CDSS\medinfo\dataconversion\test\TestFeatureMatrixFactory.py", line 281, in test_buildFeatureMatrix_prePostFeatures
    self.assertEqualList(resultMatrix, expectedMatrix)
  File "C:\HealthRex\CDSS\medinfo\common\test\Util.py", line 142, in assertEqualList
    self.assertEqual(verifyItem, sampleItem)
AssertionError: Lists differ: ['# feature-matrix_2018-04-20_... != ['pat_id', 'order_proc_id', 'p...

First differing element 0:
'# feature-matrix_2018-04-20_12.40.58.tab'
'pat_id'

Second list contains 30 additional elements.
First extra element 1:
'order_proc_id'

Diff is 825 characters long. Set self.maxDiff to None to see it.

======================================================================
FAIL: test_build_FeatureMatrix_multiLabTest (medinfo.dataconversion.test.TestFeatureMatrixFactory.TestFeatureMatrixFactory)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\HealthRex\CDSS\medinfo\dataconversion\test\TestFeatureMatrixFactory.py", line 345, in test_build_FeatureMatrix_multiLabTest
    self.assertEqualList(resultMatrix, expectedMatrix)
  File "C:\HealthRex\CDSS\medinfo\common\test\Util.py", line 142, in assertEqualList
    self.assertEqual(verifyItem, sampleItem)
AssertionError: Lists differ: ['# feature-matrix_2018-04-20_... != ['pat_id', 'order_proc_id', 'p...

First differing element 0:
'# feature-matrix_2018-04-20_12.40.59.tab'
'pat_id'

Second list contains 49 additional elements.
First extra element 1:
'order_proc_id'

Diff is 1199 characters long. Set self.maxDiff to None to see it.

======================================================================
FAIL: test_TabDictReader (medinfo.db.test.TestResultsFormatter.TestResultsFormatter)
Verify expected results when reading from different delimited file examples,
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\HealthRex\CDSS\medinfo\db\test\TestResultsFormatter.py", line 48, in test_TabDictReader
    self.assertEqualDictList( expectedData, parsedData, targetKeys );
  File "C:\HealthRex\CDSS\medinfo\common\test\Util.py", line 203, in assertEqualDictList
    self.assertEqualDict(verifyItem, sampleItem, targetKeys);
  File "C:\HealthRex\CDSS\medinfo\common\test\Util.py", line 195, in assertEqualDict
    self.assertEquals(verifyDict[key], sampleDict[key], msg % (key, str(verifyDict[key]), str(sampleDict[key])));
AssertionError: Dicts diff with key (display_name).  Verify = Give patient "Bedside Insulin Pump Flow Sheet" to document insulin delivery, BG and carbohydrates, Sample = Give patient Bedside Insulin Pump Flow Sheet" to document insulin delivery

----------------------------------------------------------------------
Ran 120 tests in 97.907s

FAILED (failures=6, errors=2)
.............................................................................................................................................................EFFFFF............................................F....E
======================================================================
ERROR: medinfo.dataconversion.test.TestEventDigraph (unittest.loader.ModuleImportFailure)
----------------------------------------------------------------------
ImportError: Failed to import test module: medinfo.dataconversion.test.TestEventDigraph
Traceback (most recent call last):
  File "C:\Dev\Python27\lib\unittest\loader.py", line 254, in _find_tests
    module = self._get_module_from_name(name)
  File "C:\Dev\Python27\lib\unittest\loader.py", line 232, in _get_module_from_name
    __import__(name)
  File "C:\HealthRex\CDSS\medinfo\dataconversion\test\TestEventDigraph.py", line 6, in <module>
    import networkx as nx
ImportError: No module named networkx

======================================================================
ERROR: test_newExtractorOutput (scripts.LabTestAnalysis.machine_learning.dataExtraction.testNewExtractorOutput.TestOutput)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\HealthRex\CDSS\scripts\LabTestAnalysis\machine_learning\dataExtraction\testNewExtractorOutput.py", line 18, in test_newExtractorOutput
    firstFile = gzip.open("labFeatureMatrix.LABFER.2013-01-01.2013-01-02.tab.gz",'r').read(12000)
  File "C:\Dev\Python27\lib\gzip.py", line 34, in open
    return GzipFile(filename, mode, compresslevel)
  File "C:\Dev\Python27\lib\gzip.py", line 94, in __init__
    fileobj = self.myfileobj = __builtin__.open(filename, mode or 'rb')
IOError: [Errno 2] No such file or directory: 'labFeatureMatrix.LABFER.2013-01-01.2013-01-02.tab.gz'

======================================================================
FAIL: test_addTimeCycleFeatures (medinfo.dataconversion.test.TestFeatureMatrixFactory.TestFeatureMatrixFactory)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\HealthRex\CDSS\medinfo\dataconversion\test\TestFeatureMatrixFactory.py", line 447, in test_addTimeCycleFeatures
    self.assertEqualList(resultMatrix, expectedMatrix)
  File "C:\HealthRex\CDSS\medinfo\common\test\Util.py", line 142, in assertEqualList
    self.assertEqual(verifyItem, sampleItem)
AssertionError: Lists differ: ['# feature-matrix_2018-04-20_... != ['pat_id', 'order_proc_id', 'p...

First differing element 0:
'# feature-matrix_2018-04-20_12.42.32.tab'
'pat_id'

Second list contains 10 additional elements.
First extra element 1:
'order_proc_id'

- ['# feature-matrix_2018-04-20_12.42.32.tab']
+ ['pat_id',
+  'order_proc_id',
+  'proc_code',
+  'order_time',
+  'normal_results',
+  'order_time.month',
+  'order_time.month.sin',
+  'order_time.month.cos',
+  'order_time.hour',
+  'order_time.hour.sin',
+  'order_time.hour.cos']

======================================================================
FAIL: test_buildFeatureMatrix_multiClinicalItem (medinfo.dataconversion.test.TestFeatureMatrixFactory.TestFeatureMatrixFactory)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\HealthRex\CDSS\medinfo\dataconversion\test\TestFeatureMatrixFactory.py", line 227, in test_buildFeatureMatrix_multiClinicalItem
    self.assertEqualList(resultMatrix, expectedMatrix)
  File "C:\HealthRex\CDSS\medinfo\common\test\Util.py", line 142, in assertEqualList
    self.assertEqual(verifyItem, sampleItem)
AssertionError: Lists differ: ['# feature-matrix_2018-04-20_... != ['pat_id', 'order_proc_id', 'p...

First differing element 0:
'# feature-matrix_2018-04-20_12.42.33.tab'
'pat_id'

Second list contains 56 additional elements.
First extra element 1:
'order_proc_id'

Diff is 1516 characters long. Set self.maxDiff to None to see it.

======================================================================
FAIL: test_buildFeatureMatrix_multiFlowsheet (medinfo.dataconversion.test.TestFeatureMatrixFactory.TestFeatureMatrixFactory)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\HealthRex\CDSS\medinfo\dataconversion\test\TestFeatureMatrixFactory.py", line 405, in test_buildFeatureMatrix_multiFlowsheet
    self.assertEqualList(resultMatrix, expectedMatrix)
  File "C:\HealthRex\CDSS\medinfo\common\test\Util.py", line 142, in assertEqualList
    self.assertEqual(verifyItem, sampleItem)
AssertionError: Lists differ: ['# feature-matrix_2018-04-20_... != ['pat_id', 'order_proc_id', 'p...

First differing element 0:
'# feature-matrix_2018-04-20_12.42.34.tab'
'pat_id'

Second list contains 49 additional elements.
First extra element 1:
'order_proc_id'

Diff is 1559 characters long. Set self.maxDiff to None to see it.

======================================================================
FAIL: test_buildFeatureMatrix_prePostFeatures (medinfo.dataconversion.test.TestFeatureMatrixFactory.TestFeatureMatrixFactory)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\HealthRex\CDSS\medinfo\dataconversion\test\TestFeatureMatrixFactory.py", line 281, in test_buildFeatureMatrix_prePostFeatures
    self.assertEqualList(resultMatrix, expectedMatrix)
  File "C:\HealthRex\CDSS\medinfo\common\test\Util.py", line 142, in assertEqualList
    self.assertEqual(verifyItem, sampleItem)
AssertionError: Lists differ: ['# feature-matrix_2018-04-20_... != ['pat_id', 'order_proc_id', 'p...

First differing element 0:
'# feature-matrix_2018-04-20_12.42.34.tab'
'pat_id'

Second list contains 30 additional elements.
First extra element 1:
'order_proc_id'

Diff is 825 characters long. Set self.maxDiff to None to see it.

======================================================================
FAIL: test_build_FeatureMatrix_multiLabTest (medinfo.dataconversion.test.TestFeatureMatrixFactory.TestFeatureMatrixFactory)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\HealthRex\CDSS\medinfo\dataconversion\test\TestFeatureMatrixFactory.py", line 345, in test_build_FeatureMatrix_multiLabTest
    self.assertEqualList(resultMatrix, expectedMatrix)
  File "C:\HealthRex\CDSS\medinfo\common\test\Util.py", line 142, in assertEqualList
    self.assertEqual(verifyItem, sampleItem)
AssertionError: Lists differ: ['# feature-matrix_2018-04-20_... != ['pat_id', 'order_proc_id', 'p...

First differing element 0:
'# feature-matrix_2018-04-20_12.42.35.tab'
'pat_id'

Second list contains 49 additional elements.
First extra element 1:
'order_proc_id'

Diff is 1199 characters long. Set self.maxDiff to None to see it.

======================================================================
FAIL: test_TabDictReader (medinfo.db.test.TestResultsFormatter.TestResultsFormatter)
Verify expected results when reading from different delimited file examples,
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\HealthRex\CDSS\medinfo\db\test\TestResultsFormatter.py", line 48, in test_TabDictReader
    self.assertEqualDictList( expectedData, parsedData, targetKeys );
  File "C:\HealthRex\CDSS\medinfo\common\test\Util.py", line 203, in assertEqualDictList
    self.assertEqualDict(verifyItem, sampleItem, targetKeys);
  File "C:\HealthRex\CDSS\medinfo\common\test\Util.py", line 195, in assertEqualDict
    self.assertEquals(verifyDict[key], sampleDict[key], msg % (key, str(verifyDict[key]), str(sampleDict[key])));
AssertionError: Dicts diff with key (display_name).  Verify = Give patient "Bedside Insulin Pump Flow Sheet" to document insulin delivery, BG and carbohydrates, Sample = Give patient Bedside Insulin Pump Flow Sheet" to document insulin delivery

----------------------------------------------------------------------
Ran 120 tests in 96.432s

FAILED (failures=6, errors=2)

**Comment by SanthoshBala** [comment]
Walking through the modules that failed:
• TestEventDigraph – test for some custom event visualizations I made. Failed just because you don't have the requisite libraries installed, so going to ignore for now.
• TestNewExtractorOutput – only relevant to the old DataExtractor code, and just cluttering the directory, so just removed from the repo.
• TestFeatureMatrixFactory – flaky test due to floating point precision. Fixed, and will convert to fuzzy equality checks if it's flaky again.
• TestResultsFormatter – This test clearly fails, but it's not obvious to me that it ever worked, so retitling this task
